shelley/chain-and-ledger/executable-spec/src/Keys.hs

```diff
@@ -240,3 +250,35 @@ hashKeyES
   -> KeyHashES hashAlgo kesAlgo
 hashKeyES (VKeyES vKeyES) =
   KeyHashES $ hashWithSerialiser encodeVerKeyKES vKeyES
+
+-- | Our VRFAlgorithm provides a 'Natural', so we must exhibit an isomorphism to
+--   use the bits as some other type
+class VRFValue a where
+  fromNatural :: Natural -> a
+  toNatural :: a -> Natural
+
+instance VRFValue Seed where
+  fromNatural = \case
+    0 -> NeutralSeed
+    1 -> SeedL
+    2 -> SeedEta
+    n -> Nonce (n - 3)
+
+  toNatural = \case
+    Nonce n -> n + 3
+    NeutralSeed -> 0
+    SeedL -> 1
+    SeedEta -> 2
+    SeedOp s1 s2 -> toNatural s1 `xor` toNatural s2
+
+instance VRFValue UnitInterval where
+  fromNatural = undefined
+  toNatural = undefined
+  -- toSeed u = mkNonce $ (numerator r * denominator r)
```

Comment:
ok, so I am a not sure how to convert for `UnitInterval` in a general way, we might want to specialize the type of unitinterval for use as leader election value. Maybe create an issue for this